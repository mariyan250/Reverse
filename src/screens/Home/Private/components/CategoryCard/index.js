// React and Style
import React from 'react';
import './index.scss';

import { Link } from 'react-router-dom';

function CategoryCard(props) {
  return (
    <section {...props}>
      <article className="category-card">
        <Link
          to={props.url}
          className="category-link d-flex flex-column justify-content-center align-items-center h-100 text-decoration-none"
        >
          <img src={props.icon} className="mt-2" alt="" />
          <p className="position-relative">{props.title}</p>
        </Link>
      </article>
    </section>
  );
}

export default CategoryCard;
