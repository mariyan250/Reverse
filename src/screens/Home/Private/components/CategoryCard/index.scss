@import '../../../../../index.scss';

.category-card {
  height: 10rem;
  box-shadow: 1px 1px 5px $silver;

  .category-link {
    color: $text-black;

    i {
      font-size: 52px;
    }

    p {
      top: 2rem;
      font-size: 16px;
    }
  }
}
