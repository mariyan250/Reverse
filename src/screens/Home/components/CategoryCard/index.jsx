// React and Style
import React from 'react';
import styles from './index.module.scss';

// Translation
import { useTranslation } from 'react-i18next';

// Router
import { Link } from 'react-router-dom';

function CategoryCard({ url, icon, title }) {
  const { t } = useTranslation();

  return (
    <Link to={url} className={`${styles.card} d-block box-shadow text-center py-2 py-md-4`}>
      <img src={icon} className="mt-2" alt={t('home.categories.img_alt')} />
      <p className="app-text-secondary mt-3 mb-0">{t(title)}</p>
    </Link>
  );
}

export default CategoryCard;
