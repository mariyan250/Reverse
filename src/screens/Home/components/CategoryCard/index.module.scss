@import '../../../../index.scss';

.card {
  img {
    width: to-rem(48px);
  }

  p {
    @include media-breakpoint-md-up {
      font-size: to-rem(16px);
    }
  }
}
