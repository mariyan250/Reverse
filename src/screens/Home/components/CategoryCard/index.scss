@import '../../../../index.scss';

.category-card {
  img {
    width: to-rem(56px);
  }

  p {
    @include media-breakpoint-md-up {
      font-size: to-rem(16px);
    }
  }
}
