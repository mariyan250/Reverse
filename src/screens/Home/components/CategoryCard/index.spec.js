import React from 'react';
import { withWrapper } from 'tests/test-utils';
import { create } from 'react-test-renderer';
import CategoryCard from './index';

describe('CategoryCard', () => {
  let initialState;

  beforeAll(() => {
    initialState = {};
  });

  describe('Rendering', () => {
    it('should render the component with initial state', () => {
      const component = create(withWrapper(<CategoryCard url="/" />, initialState));
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
