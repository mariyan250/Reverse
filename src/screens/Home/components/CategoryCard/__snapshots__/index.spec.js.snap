// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryCard Rendering should render the component with initial state 1`] = `
<a
  className="card py-2 py-md-4 box-shadow d-block text-center"
  href="/"
  onClick={[Function]}
>
  <img
    alt="home_screen.private.categories.img_alt"
    className="mt-2"
  />
  <p
    className="mt-3 mb-0 app-text-secondary"
  >
    
  </p>
</a>
`;
